functions.php

```diff
@@ -349,6 +349,24 @@ function ($urls, $post_id) {
     2
 );
 
+// Add filters to the News & Stories listing page.
+// Right now only "category" is available.
+add_action(
+    'pre_get_posts',
+    function ($query): void {
+        if (!$query->is_main_query() || is_admin() || !is_home()) {
+            return;
+        }
+        $category_slug = isset($_GET['category']) ? $_GET['category'] : '';
+        if (!empty($category_slug)) {
+            $category = get_category_by_slug($category_slug);
+            $category_id = $category->term_id;
+        }
+        $category_filter = empty($category_slug) ? [] : [$category_id];
```

Comment:
You don’t need to initialize `$category_filter` if the category is empty. You can directly set it within the if statement to improve readability.

Reply:
Done